font/src/ft/fc/pattern.rs

```diff
@@ -528,7 +528,7 @@ impl PatternRef {
         unsafe { self.add_integer(b"rgba\0", rgba.to_isize()) }
     }
 
-    pub fn render_prepare(&self, config: &ConfigRef, request: &PatternRef) -> Pattern {
+    pub fn render_prepare(&self, config: &ConfigRef, request: &PatternRef) -> self::Pattern {
```

Comment:
There's three instances of `self::Pattern` in this file, however the module itself is `pattern.rs`. Shouldn't the `Pattern` be defined in this file, thus not needing any prefix at all?

Reply:
I thought so as well, and tried to figure out what the problem was - before I gave up. Rustc is clever enough that the prefix is unneeded for compilation. but I have found RLS and IntelliJ to be much less clever about macro expansion (my guess at the root cause, the foreign_type macro). This was the case on both Windows and Debian, stable and nightly. So should I go with the syntactically correct original, or the practical existing submission?

Reply:
We do not need to cater to RLS and IntelliJ, our code is compiled by rustc. So if the IDEs and editors don't understand it, that's their problem not ours.

Reply:
Ok, I'll revert the catering.